### PR TITLE
Emit error when querying native histograms with extended functions

### PR DESCRIPTION
**Pull request by pedro-stanaka**
We saw panics on our setup because users tried to use the extended functions with native histograms, which led to rates calculated with only one sample in some windows.

In this PR I am introducing a safeguard against this type of situation and emiting an error when users try to query native histograms with extended functions.

### CODE REVIEW

**Review by fpetkovski — APPROVED** [review]

File: execution/scan/matrix_selector.go [hunk 1]
```diff
@@ -351,8 +353,9 @@ loop:
 // into the [mint, maxt] range are retained; only points with later timestamps
 // are populated from the iterator.
 // TODO(fpetkovski): Add max samples limit.
-func selectExtPoints(it *storage.BufferedSeriesIterator, mint, maxt int64, out []Sample, extLookbackDelta int64, metricAppearedTs **int64) ([]Sample, error) {
+func selectExtPoints(ctx context.Context, it *storage.BufferedSeriesIterator, mint, maxt int64, out []Sample, extLookbackDelta int64, metricAppearedTs **int64) ([]Sample, error) {
```
Comment (GiedriusS) [inline execution/scan/matrix_selector.go:356]: Why this context was added here? Seems like an unrelated change

Comment (pedro-stanaka) [reply execution/scan/matrix_selector.go]: Good catch, I was using the context before to attach warns, but we saw it was better to return an error altogether.

**Comment by pedro-stanaka** [comment]
Seems like we got a flaky test, `make test` on my local succeeds, but not on CI. Trying an empty commit to see if succeeds.

**Review by GiedriusS — APPROVED** [review]
